src/Tools/BuildValidator/LocalSourceResolver.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+
+namespace BuildValidator
+{
+    /// <summary>
+    /// Roslyn specific implementation for looking for files
+    /// in the Roslyn repo
+    /// </summary>
+    internal class LocalSourceResolver : ISourceResolver
+    {
+        private readonly DirectoryInfo _baseDirectory;
+        private readonly ILogger _logger;
+
+        public LocalSourceResolver(ILoggerFactory loggerFactory)
+        {
+            _baseDirectory = GetSourceDirectory();
+            _logger = loggerFactory.CreateLogger<LocalSourceResolver>();
+
+            _logger.LogInformation($"Source Base Directory: {_baseDirectory}");
+        }
+
+        public Task<SourceText> ResolveSourceAsync(string name, Encoding encoding)
```

Comment:
If all implementations of this are sync then why are we using a `Task` return here?

Reply:
Changed to synchronous implementation